device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubListener.java

```diff
@@ -43,4 +43,16 @@
      * @param connectionId the id of the connection this update is relevant to
      */
     void onConnectionEstablished(String connectionId);
+
+    /**
+     * Callback to be fired when the multiplexed connection establishes a new device session.
+     * @param deviceId the Id of the device that the session belongs to
+     */
+    void onDeviceSessionEstablished(String connectionId, String deviceId);
```

Comment:
This can have `multiplex` in the name of the call back to avoid confusion ... maybe `onDeviceMultiplexSessionEstablished`
same for the method below.